### ISSUE TITLE
Missing `'ignore'` in filename decoding

### ISSUE THREAD

**Issue by claugner** [erev0s/apkInspector#27]
### Describe what you wanted to do
Ignore non-utf8 characters in filename or handle them properly.

### Describe what you expected
Non-utf8 characters in filename being ignored or properly decoded. For example, the filenames in the [sample](https://github.com/Chuyu-Team/Dism-Multi-language/releases/download/v10.1.1002.2/Dism++10.1.1002.1B.zip) (although not an APK) is encoded in GBK and `zipfile` decodes them properly.

### Describe what actually happened
  [File "/foo/bar/apkInspector/headers.py", line 239](https://github.com/erev0s/apkInspector/blob/6ed2038fb4e0e770b4ce93b72f833836110ab904/apkInspector/headers.py#L239C1-L239C114), in parse
    filename = struct.unpack(f'<{file_name_length}s', apk_file.read(file_name_length))[0].decode('utf-8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xbb in position 2: invalid start byte

### System Information

- apkInspector Version/Branch: v1.2.4

**Comment by erev0s** [comment]
hi @claugner , thank you for reporting this. I have updated it [here](https://github.com/erev0s/apkInspector/commit/bc71f155526b6abe87f410742dcd663fb0275f13) to ignore non-utf8 characters for a cleaner output.

**Comment by obfusk** [comment]
FYI:

* it looks like the decoding for local header records still doesn't use `ignore`
* there's [a flag for UTF-8 filenames in ZIP files and the default encoding if it's not set is `cp437`](https://github.com/python/cpython/blob/main/Lib/zipfile/__init__.py#L1483)
* IIRC some APKs will have the flag unset even though they use UTF-8 so for tools like this is does seem a good idea to not blindly trust the flag is set correctly

**Comment by erev0s** [comment]
@obfusk thank you ! always a keen eye <3
I have updated it, although it would not error out as if it did then the `except`  clause in there would substitute the filename with the value from the central directory, but for completeness I believe it is better if it is there.

my only question initially was whether it might had any advantage if instead of ignoring it, to replace it. But I went for the ignoring as it would provide a cleaner output and its not like we would be missing important information.
